flux_local/values.py

```diff
@@ -173,6 +173,19 @@ def _lookup_value_reference(
 
     return found_value
 
+def deep_merge(base: dict, override: dict) -> dict:
+    """
+    Recursively merge two dictionaries, similar to how Helm merges values.
+    Lists are replaced entirely (Helm behavior).
+    """
```

Comment:
Style in this code base is:

```suggestion
    """Recursively merge two dictionaries, similar to how Helm merges values.

    Lists are replaced entirely (Helm behavior).
    """
```